### PR TITLE
[DNM] wip: qa: add basic rgw ssl deployment tests

### PR DESCRIPTION
**Pull request by theanalyst**
Creates the rgw ssl configuration and does a curl test with no ssl
verification

Signed-off-by: Abhishek Lekshmanan <abhishek@suse.com>

### CODE REVIEW

**Comment by theanalyst** [comment]
Have a nearly final version of the script, it works for me locally, currently depends on https://github.com/SUSE/DeepSea/pull/530

**Comment by smithfarm** [comment]
@theanalyst Can you give me a wip branch that has #530 and this PR? I could do a teuthology run on it.

**Comment by smithfarm** [comment]
Oh. . . nevermind. Scheduling.

File: qa/suites/basic/health-rgw-ssl.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+#
+# DeepSea integration test "suites/basic/health-rgw.sh"
+#
+# This script runs DeepSea stages 0-4 to deploy a Ceph cluster with RGW.  After
+# stage 4 completes, it sends a GET request to the RGW node using curl, and
+# tests that: (a) the response contains the string "anonymous" and (b) the
+# response is legal XML.
```
Comment (smithfarm) [inline qa/suites/basic/health-rgw-ssl.sh]: This looks like the same description as for `health-rgw.sh` . . . can it be updated?

Comment (theanalyst) [inline qa/suites/basic/health-rgw-ssl.sh]: agree, had just copy-pasted, will fix this

File: qa/common/common.sh [hunk 2]
```diff
@@ -174,6 +174,16 @@ role-rgw/cluster/*.sls slice=[:1]
...
 
+function policy_cfg_rgw_ssl {
+    cat <<EOF >> /srv/pillar/ceph/proposals/policy.cfg
+# Role assignment - rgw (first node)
+role-rgw/cluster/*.sls slice=[:1]
+role-rgw-ssl/cluster/*.sls slice=[:1]
+EOF
+}
+
+
```
Comment (theanalyst) [inline qa/common/common.sh]: too many blank lines

File: Makefile [hunk 3]
```diff
@@ -82,6 +82,7 @@ copy-files:
 	install -m 644 srv/salt/ceph/configuration/files/*.j2 $(DESTDIR)/srv/salt/ceph/configuration/files/
 	install -m 644 srv/salt/ceph/configuration/files/*.rbd $(DESTDIR)/srv/salt/ceph/configuration/files/
 	install -m 644 srv/salt/ceph/configuration/files/*.rgw $(DESTDIR)/srv/salt/ceph/configuration/files/
+	install -m 644 srv/salt/ceph/configuration/files/*.rgw-ssl $(DESTDIR)/srv/salt/ceph/configuration/files/
```
Comment (theanalyst) [inline Makefile]: Maybe worth pulling into as a seperate PR

File: qa/common/rgw.sh [hunk 4]
```diff
@@ -51,3 +51,52 @@ EOF
...
+    local TESTSCRIPT=/tmp/rgw_test.sh
+    cat << 'EOF' > $TESTSCRIPT
+set -ex
+trap 'echo "Result: NOT_OK"' ERR
+echo "rgw curl test running as $(whoami) on $(hostname --fqdn)"
+RGWNODE=$(salt --no-color -C "I@roles:rgw" test.ping | grep -o -P '^\S+(?=:)' | head -1)
+zypper --non-interactive --no-gpg-checks refresh
+zypper --non-interactive install --no-recommends curl libxml2-tools
+RGWXMLOUT=/tmp/rgw_test.xml
+curl -k https://$RGWNODE  > $RGWXMLOUT
```
Comment (theanalyst) [inline qa/common/rgw.sh]: the only difference between this fun and its predecessor is the -k & https:// bit, maybe having another helper function might solve this

Comment (smithfarm) [inline qa/common/rgw.sh]: I'm fine with it as it is.

**Comment by smithfarm** [comment]
The new test passes in teuthology, too: http://149.202.189.249:8081/ubuntu-2017-08-11_19:42:25-deepsea:basic:rgw-ssl-ses5---basic-openstack/

File: qa/common/rgw.sh (hunk 4 above)
Comment (theanalyst) [inline qa/common/rgw.sh]: ok, I'll keep it as it is, will just get the blank line in the earlier comment once #530 goes in

**Pull request merged**